oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleSource.java

```diff
@@ -153,6 +154,17 @@ public void validate(FailureCollector collector) {
     public String getTransactionIsolationLevel() {
       return connection.getTransactionIsolationLevel();
     }
+
+    @Override
+    protected void validateField(FailureCollector collector,
+                                 Schema.Field field, Schema actualFieldSchema, Schema expectedFieldSchema) {
+      // Validate if the field output schema is set to Decimal and the actual evaluated schema is String.
+      if (Schema.LogicalType.DECIMAL.equals(expectedFieldSchema.getLogicalType())
+          && actualFieldSchema.getType().equals(Schema.Type.STRING)) {
+        return;
+      }
```

Comment:
Why is this change needed? When do we expect actual schema (schema in user's database) as `STRING` and output schema to be `DECIMAL`?

Reply:
This change is needed to make sure that the pipeline upgrade continues to work post upgrade. Since the older handling of the precision less used to convert to the decimal type, and the new version would try to convert to the String type. In that case the output schema would contain Decimal(38, 0) (or something similar), and the code internally would try to identify the schema of the field(without precision and scale) as String.

Reply:
would be good to put this explanation in the code comment